UnchainedLauncher.Core/Services/PakDir/PakDir.cs

```diff
@@ -0,0 +1,364 @@
+using LanguageExt;
+using LanguageExt.Common;
+using log4net;
+using UnchainedLauncher.Core.Extensions;
+using UnchainedLauncher.Core.Services.Mods.Registry;
+using UnchainedLauncher.Core.Utilities;
+
+namespace UnchainedLauncher.Core.Services.PakDir {
+    using static LanguageExt.Prelude;
+
+    public class PakDir : IPakDir {
+        private readonly string _dirPath;
+        private readonly string _metadataFileName;
+
+        private const string BasePakFileName = "pakchunk0-WindowsNoEditor.pak";
+        private const string BaseSigFileName = "pakchunk0-WindowsNoEditor.sig";
+
+        private static readonly ILog Logger = LogManager.GetLogger(nameof(PakDir));
+
+        // TODO: invert this map, because most uses of it are filtering on the Value, not the key.
+        private Map<string, ReleaseCoordinates> _releaseMap;
```

Comment:
Might make sense to have both Map<string, ReleaseCoordinates> and Map<ReleaseCoordinates, string>, assuming its immutable